pkg/controller/gitopsservice/dependency/dependency.go

```diff
@@ -0,0 +1,269 @@
+	err := d.client.Get(ctx, ns, obj)
+	if err != nil {
+		switch errors.ReasonForError(err) {
+		case metav1.StatusReasonNotFound:
+			err = d.client.Create(ctx, obj)
+			if err != nil {
+				d.log.Error(err, "Unable to create resource", "Resource.Kind", obj.GetObjectKind(), "Resource.Name", ns.
+					Name)
+				return err
+			}
+			d.log.Info("Successfully created resource", "Resource.Kind", obj.GetObjectKind(), "Resource.Name", ns.Name, "Resource.Namespace", ns.
+				Namespace)
+		case metav1.StatusReasonAlreadyExists:
+			d.log.Info("Resource already exists", "Resource.Kind", obj.GetObjectKind(), "Resource.Name", ns.Name)
+		default:
+			return err
+		}
+	}
+	return nil
+}
+
+func newNamespace(name string) *corev1.Namespace {
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+}
+
+func newOperatorGroup(namespace, name string) *v1.OperatorGroup {
+	return &v1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1.OperatorGroupSpec{
+			TargetNamespaces: []string{namespace},
+		},
+	}
+}
+
+func newSubscription(namespace, name string) *v1alpha1.Subscription {
+	return &v1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: &v1alpha1.SubscriptionSpec{
+			Channel:                "alpha",
+			CatalogSource:          "community-operators",
+			CatalogSourceNamespace: "openshift-marketplace",
+			Package:                name,
+		},
+	}
+}
+
+func addPrefixIfNecessary(prefix, name string) string {
+	if prefix != "" {
+		return prefix + "-" + name
+	}
+	return name
+}
+
+func argoCDCr(ns string) (runtime.Object, string, error) {
+	name := "argocd"
+	b, err := yaml.Marshal([]resource{
+		{
+			APIGroups: []string{"tekton.dev"},
+			Kinds:     []string{"TaskRun", "PipelineRun"},
+			Clusters:  []string{"*"},
+		},
+	})
+	if err != nil {
+		return nil, "", err
+	}
+	return &argoapp.ArgoCD{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ArgoCD",
+			APIVersion: "argoproj.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+		Spec: argoapp.ArgoCDSpec{
+			ResourceExclusions: string(b),
+			Server: argoapp.ArgoCDServerSpec{
+				Route: argoapp.ArgoCDRouteSpec{Enabled: true},
+			},
+		},
+	}, name, nil
+}
+
+func sealedSecretsCR(ns string) (runtime.Object, string, error) {
+	name := "sealedsecretcontroller"
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "bitnami.com/v1alpha1",
+			"kind":       "SealedSecretController",
+			"metadata": map[string]interface{}{
+				"name":      name,
+				"namespace": ns,
+			},
+			"spec": map[string]interface{}{
+				"image": map[string]interface{}{
+					"repository": "quay.io/bitnami/sealed-secrets-controller@sha256:8e9a37bb2e1a6f3a8bee949e3af0e9dab0d7dca618f1a63048dc541b5d554985",
```

Comment:
We need to be able get image from environment variable in order to support disconnected install.

Reply:
Added the support to read images from env variable